### PR TITLE
Add tests and doc to sdist

### PR DESCRIPTION
**Pull request by maresb**
This makes it more viable to use the PyPI sdist for building Gentoo, as per https://github.com/aesara-devs/aesara/discussions/1453#discussioncomment-5092297

**Thank you for opening a PR!**

Here are a few important guidelines and requirements to check before your PR can be merged:
+ [X] There is an informative high-level description of the changes.
+ [X] The description and/or commit message(s) references the relevant GitHub issue(s).
+ [X] [`pre-commit`](https://pre-commit.com/#installation) is installed and [set up](https://pre-commit.com/#3-install-the-git-hook-scripts).
+ [X] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [X] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes), and there are **no commits that fix changes introduced by other commits in the same branch/BR**.
+ [ ] There are tests covering the changes introduced in the PR.

Don't worry, your PR doesn't need to be in perfect order to submit it.  As development progresses and/or reviewers request changes, you can always [rewrite the history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) of your feature/PR branches.

If your PR is an ongoing effort and you would like to involve us in the process, simply make it a [draft PR](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).

### CODE REVIEW

**Comment by maresb** [comment]
Reverts #1408, but if people find it useful to include the docs in the sdist then we might as well.

File: pyproject.toml [hunk 1]
```diff
@@ -77,15 +77,13 @@ version-file = "aesara/_version.py"
...
     "/.flake8",
     "/.gitignore",
     "/.hgignore",
     "/.pre-commit-config.yaml",
     "/CODE_OF_CONDUCT.md",
     "/CONTRIBUTING.md",
-    "/doc",
     "Makefile",
     "codecov.yml",
     "conftest.py",
```
Comment (mgorny) [inline pyproject.toml:89]:
```suggestion
```

This one's used by pytest too.

**Review by mgorny — COMMENTED** [review]
Thanks!

**Comment by maresb** [comment]
Let's squash-merge when ready

**Pull request merged**